### PR TITLE
[BTS-152] Alert component 

### PR DESCRIPTION
**Pull request by alan-oliv**
New alert component according UX definitions

### CODE REVIEW

**Comment by rapahaeru** [comment]
In active mode, the close button shows a background-color with radius.
Was this behavior predicted?

Captura de tela de 2019-03-22 14-55-24

**Comment by rapahaeru** [comment]
as this story updated the button component, i have noticed that this warning shows on console when i check and uncheck any checkbox.
has this behavior already existed? 

Captura de tela de 2019-03-22 15-40-09

File: components/Alert/Alert.jsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  padding: ${WRAPPER_SPACING} ${DEFAULT_SPACING};
+
+  ${({ skin }) => getStylesBySkin(skin)}
+`;
+
+CloseButton.displayName = 'CloseButton';
+
+const Alert = ({ icon, children, onClose, ...rest }) => (
+  <Wrapper {...rest}>
+    <Content onClose={onClose}>
```
Comment (rapahaeru) [inline components/Alert/Alert.jsx]: what do you think about put a aria `role="alert"` here to assistive technologies?

https://developer.paciellogroup.com/blog/2012/06/html5-accessibility-chops-aria-rolealert-browser-support/

File: components/Alert/Alert.unit.test.jsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import Alert from './Alert';
+
+describe('Alert component', () => {
```
Comment (rapahaeru) [inline components/Alert/Alert.unit.test.jsx]: I believe that this tests cover are suficient.
It you want to create a redundancy coverage, i would suggest a new two tests:
- verify if the children text passed is the same;
- verify when the callback onClose was set, if the icon close ir really is the close icon;

WIth Daniel in the input and textarea stories, we did it that way.
For a children case, for example:
```
    it('should the children text be the same at the prop', () => {
      const childrenContent = 'lorem ipsum ';
      const component = mount(
        <Alert onClose={() => {}}>{childrenContent}</Alert>,
      );
      const componentChildrenContent = component
        .find('Content')
        .childAt(0)
        .text();

      expect(componentChildrenContent).toMatch(childrenContent);
    });
```
I dont know if i was clear rsss

**Comment by alan-oliv** [comment]
> In active mode, the close button shows a background-color with radius.
> Was this behavior predicted?
> 
> Captura de tela de 2019-03-22 14-55-24

This was not predicted on Zeplin, therefore, it was a decision we made - user must have a :hover feedback, what do you think?

File: components/Alert/Alert.unit.test.jsx (hunk 2 above)
Comment (alan-oliv) [reply components/Alert/Alert.unit.test.jsx]: Wow! Nice!

File: components/Alert/Alert.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    ${Content} ${CloseButton} {
+      color: ${Colors[colorSchema][500]};
+    }
+  `;
+};
+
+const Wrapper = styled.div`
+  border-radius: 8px;
+  box-sizing: border-box;
+  padding: ${WRAPPER_SPACING} ${DEFAULT_SPACING};
```
[the remark this answers is not in the document]
Comment (alan-oliv) [reply components/Alert/Alert.jsx]: This is intentional...
When we use 16px on padding-top and bottom, it sums the span line-height and the eye perception is that top and bottom padding are greater than sides... that was intentional

File: components/Alert/Alert.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  ${({ skin }) => getStylesBySkin(skin)}
+`;
+
+CloseButton.displayName = 'CloseButton';
```
[the remark this answers is not in the document]
Comment (alan-oliv) [reply components/Alert/Alert.jsx]: Thanks for pointing me that <3

**Comment by alan-oliv** [comment]
> as this story updated the button component, i have noticed that this warning shows on console when i check and uncheck any checkbox.
> has this behavior already existed?
> 
> Captura de tela de 2019-03-22 15-40-09

Nice observation, man!
Although, this is a issue on quantum-storybook-ui, I'll create a fix for that as soon as possible!

File: components/Alert/Alert.jsx (hunk 1 above)
Comment (alan-oliv) [reply components/Alert/Alert.jsx]: I think it's great! consider it done!

File: components/Alert/Alert.jsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    background: none;
+    opacity: 1;
+  }
+`;
+
+const getStylesBySkin = skin => {
+  const colorSchema = skin.toUpperCase();
+
+  return `
+    color: ${Colors[colorSchema][900]};
```
[the remark this answers is not in the document]
Comment (alan-oliv) [reply components/Alert/Alert.jsx]: Nice! I wouldn't notice that! Thanks, man.

**Comment by alan-oliv** [comment]
> I believe we missed the unit tests.
> Or did you decided not to?

**Comment by alan-oliv** [comment]
> I believe we missed the unit tests.
> Or did you decided not to?

We decided to create a improvement task for that! Thanks man!

**Pull request merged**